AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -169,6 +173,32 @@ - (void)signIn {
   }
 }
 
+- (void)signOut {
+  @synchronized(self) {
+    if (![self canBeUsed]) {
+      return;
+    }
+    if ([MSAuthTokenContext sharedInstance].authToken != nil) {
```

Comment:
Also this check might be moved to the `[[MSAuthTokenContext sharedInstance] clearAuthToken]` since  `MSAuthTokenContext` is holding the token. `clearAuthToken` might return a boolean in this case.